packages/file-graph/fileGraph.test.js

```diff
@@ -0,0 +1,110 @@
+import FileGraph from './fileGraph.js'
+import FileNode from './fileNode.js'
+
+describe('FileGraph', () => {
+  let root
+
+  beforeEach(() => {
+    root = new FileNode({
+      path: '/',
+      type: 'folder',
+      discovered: true,
+    })
+  })
+
+  test('add files to graph', () => {
+    const fileGraph = new FileGraph(root)
+
+    const filesToAdd = [
+      {
+        path: '/home/user/file1.txt',
+        type: 'file',
+        discovered: true,
+      },
+      {
+        path: '/home/user/folder1',
+        type: 'folder',
+        discovered: true,
+      },
+    ]
+
+    fileGraph.add(filesToAdd)
+
+    console.log(fileGraph.root.toString())
+    expect(fileGraph.root.children.length).toBe(1)
+    expect(fileGraph.root.children[0].children[0].children.length).toBe(2)
+  })
+
+  test('remove files from graph', () => {
+    const fileGraph = new FileGraph(root)
+
+    const filesToAdd = [
+      {
+        path: '/home/user/file1.txt',
+        type: 'file',
+        discovered: true,
+      },
+      {
+        path: '/home/user/folder1',
+        type: 'folder',
+        discovered: true,
+      },
+    ]
+
+    fileGraph.add(filesToAdd)
+
+    const filesToRemove = [
+      {
+        path: '/home/user/folder1',
+        type: 'folder',
+      },
+    ]
+
+    fileGraph.remove(filesToRemove)
+
+    console.log(fileGraph.root.toString())
+    expect(fileGraph.root.children[0].children[0].children.length).toBe(1)
```

Comment:
There is also some cases to consider:
1. remove the parents those has children (should throw error or delete the children as well?)
2. if the removing file doesn't exists, you should but throw error in FileNode, and handle by ignoring it in FileGraph

Reply:
Checking the file name of the remaining file would be better